distribution/build.gradle

```diff
@@ -120,70 +110,181 @@ configure(subprojects.findAll { it.name == 'zip' || it.name == 'tar' }) {
+  }
+
+  task createEtc(type: EmptyDirTask) {
+    dir "$packagingFiles/etc/elasticsearch"
+    dirMode 0750
+  }
+
+  task createEtcScripts(type: EmptyDirTask) {
+    dependsOn createEtc
+    dir "$packagingFiles/etc/elasticsearch/scripts"
+    dirMode 0750
+  }
+
+  task fillEtc(type: Copy) {
+    dependsOn createEtc, createEtcScripts
+    with configFiles
+    into "$packagingFiles/etc/elasticsearch"
+    /* Explicitly declare the output files so this task doesn't consider itself
+      up to date when the directory is created, which it would by default. And
+      that'll happen when createEtc runs. */
+    outputs.file "$packagingFiles/etc/elasticsearch/elasticsearch.yml"
+    outputs.file "$packagingFiles/etc/elasticsearch/logging.yml"
+  }
+
+  task createPidDir(type: EmptyDirTask) {
+    dir "$packagingFiles/var/run/elasticsearch"
+  }
+  task createLogDir(type: EmptyDirTask) {
+    dir "$packagingFiles/var/log/elasticsearch"
+  }
+  task createDataDir(type: EmptyDirTask) {
+    dir "$packagingFiles/var/lib/elasticsearch"
+  }
+  task createPluginsDir(type: EmptyDirTask) {
+    dir "$packagingFiles/usr/share/elasticsearch/plugins"
+  }
+
+  /**
+   * Setup the build/packaging directory to be like the target filesystem
+   * because ospackage will use arbitrary permissions if you try to create a
+   * directory that doesn't exist on the filesystem.
+   */
+  task preparePackagingFiles {
+    dependsOn processPackagingFiles, fillEtc, createPidDir, createLogDir,
+        createDataDir, createPluginsDir
   }
-}
-task createEmptyDir << {
-  emptyDir.mkdirs()
-}
-buildRpm.dependsOn createEmptyDir
-buildDeb.dependsOn createEmptyDir
-*/
 
-/*****************************************************************************
- *                         Deb and rpm configuration                         *
- *****************************************************************************/
-configure(subprojects.findAll { it.name == 'deb' || it.name == 'rpm' }) {
   apply plugin: 'nebula.ospackage-base'
   ospackage {
-    packageName = 'elasticsearch'
-    // TODO: '-' is an illegal character in rpm version...redline croaks
-    version = '3.0.0'
+    packageName 'elasticsearch'
+    maintainer 'Elasticsearch Team <info@elastic.co>'
+    summary '''Elasticsearch is a distributed RESTful search engine built for
+the cloud. Reference documentation can be found at
+https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html and
+the 'Elasticsearch: The Definitive Guide' book can be found at
+https://www.elastic.co/guide/en/elasticsearch/guide/current/index.html
+'''.split('\n').join(' ')
+    url 'https://www.elastic.co/'
+
+    /* The version of the package can't contain -SNAPSHOT so we rip it off if
+      we see it. We'll add it back on to the file name though. */
+    version project.version.replace('-SNAPSHOT', '')
+
+    def scripts = "$project.packagingFiles/scripts"
+    preInstall file("$scripts/preinst")
```

Comment:
I would really prefer that we use the `${}` style for code in strings. Using just `$` makes it hard to tell when the variable name stops and literal string begins.